src/main/java/RangeCalculator.java

```diff
@@ -0,0 +1,47 @@
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class RangeCalculator {
+
+    public Map<String, BigDecimal> calculate(List<String> ranges) throws InterruptedException {
+        Map<Thread, MyRunnable> threadToRunnable = new HashMap<>(ranges.size());
+        Map<String, BigDecimal> rangeToResult = new HashMap<>(ranges.size());
+
+        for (String range : ranges) {
+            MyRunnable myRunnable = new MyRunnable(range);
+            Thread thread = new Thread(myRunnable);
+            threadToRunnable.put(thread, myRunnable);
+            thread.start();
```

Comment:
a slightly better practice (IMO):
do the `thread.start()` in a different loop.

in other words:
one loop for setup
one loop to start the execution.